### PR TITLE
Fix React combobox debounce

### PR DESCRIPTION
**Pull request by RobertJoonas**
### Changes

In the `PlausibleCombobox` component (used in the filter menus and the `prop_key` picker), we're currently making too many API calls to fetch selectable options. Currently, we're calling the API for every keystroke. This PR fixes that - we will only make a request once 200ms has passed from the last keystroke (reusing the `useDebouncedCallback` custom hook introduced in #4318).

Also in this PR:

* Alter the `useDebouncedEffect` custom hook to not set a delay for the initial function call. Currently the search in all breakdown modals is waiting 300ms before calling the API for the first time. This fixes that.
* Remove redundant doubled "search" state from the BreakdownModal component.

### Tests
- [x] This PR does not require tests

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: assets/js/dashboard/components/combobox.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React, { Fragment, useState, useCallback, useEffect, useRef } from 'react'
 import { Transition } from '@headlessui/react'
 import { ChevronDownIcon } from '@heroicons/react/20/solid'
-import debounce from 'debounce-promise'
```
Comment (ukutaht) [inline assets/js/dashboard/components/combobox.js:4]: Can we remove the `debounce-promise` dependency now as well?

File: assets/js/dashboard/components/combobox.js [hunk 2]
```diff
@@ -48,10 +48,26 @@ export default function PlausibleCombobox(props) {
...
+  useDebouncedEffect(() => {
+    if (isOpen) {
+      fetchOptions()
+    }
+  }, [search], 200)
+
+  const fetchOptions = useCallback(() => {
+    setLoading(true)
+
+    props.fetchOptions(search).then((loadedOptions) => {
```
Comment (macobo) [inline assets/js/dashboard/components/combobox.js:64]: I propose we use async over promise chaining here. Example: 

```javascript
const fetchOptions = useCallback(async () => {
  setLoading(true)

  const loadedOptions = await props.fetchOptions(search)

  setLoading(false)
  setHighlightedIndex(0)
  setOptions(loadedOptions)
}, [search])
```

File: assets/js/dashboard/components/combobox.js [hunk 3]
```diff
@@ -48,10 +48,26 @@ export default function PlausibleCombobox(props) {
...
-    visibleOptions.push({value: input, label: input, freeChoice: true})
+  if (props.freeChoice && search.length > 0 && options.every(option => option.value !== search)) {
+    visibleOptions.push({value: search, label: search, freeChoice: true})
   }
 
+  useDebouncedEffect(() => {
+    if (isOpen) {
+      fetchOptions()
+    }
+  }, [search], 200)
```
Comment (macobo) [inline assets/js/dashboard/components/combobox.js:59]: Why is this 200 in one case and 300 in another? 

Proposal: Let's unify and make it a exported const somewhere.

File: assets/js/dashboard/components/combobox.js (hunk 2 above)
Comment (RobertJoonas) [reply assets/js/dashboard/components/combobox.js:64]: I would leave it for now, as it's currently used all in pretty much every dashboard report. True, promise chaining might look a bit messy when there's multiple promises involved, but a single `.then()` clause doesn't hurt readability that much, I think?

File: assets/js/dashboard/components/combobox.js (hunk 3 above)
Comment (ukutaht) [reply assets/js/dashboard/components/combobox.js:59]: Good point, agreed. The 'standard' delay could even be baked in to the`useDebouncedEffect` and not be configurable.

File: assets/js/dashboard/components/combobox.js (hunk 1 above)
Comment (RobertJoonas) [reply assets/js/dashboard/components/combobox.js:4]: https://github.com/plausible/analytics/pull/4331/commits/9d9d66dde9f5aa4be0b08129a242fcee44a48a23

**Comment by RobertJoonas** [comment]
Closed in favor of https://github.com/plausible/analytics/pull/4349

**Pull request closed without merging**